tailnet/tunnel.go

```diff
@@ -91,6 +92,42 @@ func (a AgentCoordinateeAuth) Authorize(req *proto.CoordinateRequest) error {
 	return nil
 }
 
+type ClientUserCoordinateeAuth struct {
+	UserID          uuid.UUID
+	UpdatesProvider WorkspaceUpdatesProvider
+}
+
+func (a ClientUserCoordinateeAuth) Authorize(req *proto.CoordinateRequest) error {
+	if tun := req.GetAddTunnel(); tun != nil {
+		uid, err := uuid.FromBytes(tun.Id)
+		if err != nil {
+			return xerrors.Errorf("parse add tunnel id: %w", err)
+		}
+		isOwner := a.UpdatesProvider.IsOwner(a.UserID, uid)
```

Comment:
My understanding is that we don't currently have a great way to test that tunnel requests get authorized properly without some client-side abstraction for this new endpoint (i.e. updating the TAC), sot this is untested for now.